### ISSUE TITLE
junit-reporter causes process to fail when run against saucelabs

### ISSUE THREAD

**Issue by hborawski** [webdriverio/webdriverio#3775]
**Environment (please complete the following information):**
 - WebdriverIO version: **5.7.6**
 - Mode:**WDIO Testrunner**
 - If WDIO Testrunner, running sync/async: **async**
 - Node.js version:**10.15**
 - NPM version:**6.4.1**
 - Browser name and version:**Firefox latest (65)**
 - Platform name and version:**macOS 10.13**
 - Additional wdio packages used (if applicable): **@wdio/junit-reporter @wdio/sauce-service @wdio/mocha-framework @wdio/spec-reporter @wdio/local-runner @wdio/cli**

**Config of WebdriverIO**
Repo to reproduce the issue:
https://github.com/hborawski/wdio-sauce-junit-error

**Describe the bug**
When supplying the `outputDir` to `@wdio/junit-reporter`, tests run against saucelabs fail due to a command being sent to saucelabs not being recognized:
```
Error: GET /session/XXXXXXXX-XXXX-XXXX-XXXX-XXXX87a7c5b4/log/types did not match a known command
```

**To Reproduce**
Steps to reproduce the behavior in `README.md`:
https://github.com/hborawski/wdio-sauce-junit-error

**Expected behavior**
Test should pass and exit with exit code `0`

**Log**
https://gist.github.com/hborawski/8c5eb05a0371c329577bac3573af4720

**Additional context**
Supplying `junit` without options to the `reporters` does not exhibit this behavior, but only prints the results to stdout. In order to use this in CI/CD we need to be able to process the junit reports, so printing to stdout is not an option.

Happy to provide more information if needed

**Comment by naddison** [comment]
The issue with junit-reporter may be linked to https://github.com/webdriverio/webdriverio/issues/3557

**Comment by hborawski** [comment]
I think #3557 is specifically about `outputFileFormat` needing to be a function (as defined in the base reporter, vs the junit docs). That option is not exercised in this situation. 

In this issue, the tests run successfully, but the report generation fails specifically with saucelabs. The example repository I provided runs successfully against a local version of selenium.

**Comment by shashtrih** [comment]
@naddison  also another issue which relevant to junit reporter where 
outputFileFormat: function(opts) { // optional
                return `results-${opts.cid}.${opts.capabilities}.xml`
            }
where ${opts.capabilities} giving undefined value 
is it we can look into it?

**Comment by naddison** [comment]
Can you elaborate? Are your files being named with `undefined` in them?

**Comment by barnc** [comment]
I am getting this same error with IE11 running against saucelabs, browserstack and testingbot.

**Comment by naddison** [comment]
If you look at my comments on https://github.com/webdriverio/webdriverio/issues/3557 you will see that due to refactoring done in v4 -> v5 you no longer can use the "single/multi" functionality of the reporter. The property `outputFileFormat`  must only be a single function.

You can combine them in an onComplete() or after() hook using a library, or your own code.

**Comment by shashtrih** [comment]
@naddison I was generated xml but file name showing as “results-0-0.undefined.xml” so ${opts.capabilities} return undefined value ... is it we can get browsername and platform name in outputFileFormat 
Another issue we are seeing test are passed through spec reporter but junit reporter getting get/session/xxx/log/types not command due to that whole test suites get failed

**Comment by hborawski** [comment]
@naddison @christian-bromann Any insight into this? Where is the call to `log/types` happening?

**Comment by naddison** [comment]
@hborawski I'll take a look this weekend, no promises.

**Comment by naddison** [comment]
@hborawski Can you update all your wdio packages and try again? I was able to reproduce the problem using your example, but then I upgraded all the packages and have not been able to reproduce since. We might have accidentally fixed this?

**Comment by naddison** [comment]
@christian-bromann this can be closed since you merged https://github.com/webdriverio/webdriverio/pull/3870

**Comment by hborawski** [comment]
👍 Yep all good after #3870. Thanks!
